erizoAPI/WebRtcConnection.cc

```diff
@@ -510,9 +515,21 @@ NAUV_WORK_CB(WebRtcConnection::promiseResolver) {
   ELOG_DEBUG("%s, message: promiseResolver", obj->toLog());
   obj->futures_manager_.cleanResolvedFutures();
   while (!obj->futures.empty()) {
-    auto persistent = obj->futures.front();
+    auto persistent = obj->futures.front().first;
     v8::Local<v8::Promise::Resolver> resolver = Nan::New(*persistent);
-    resolver->Resolve(Nan::GetCurrentContext(), Nan::New("").ToLocalChecked());
+    ResultVariant r = obj->futures.front().second;
+    if (boost::get<std::string>(&r) != nullptr) {
+      resolver->Resolve(Nan::GetCurrentContext(), Nan::New(boost::get<std::string>(r).c_str()).ToLocalChecked());
+    } else if (boost::get<std::shared_ptr<erizo::SdpInfo>>(&r) != nullptr) {
+      std::shared_ptr<erizo::SdpInfo> sdp_info = boost::get<std::shared_ptr<erizo::SdpInfo>>(r);
+      v8::Local<v8::Object> instance = ConnectionDescription::NewInstance();
+      ConnectionDescription* description = ObjectWrap::Unwrap<ConnectionDescription>(instance);
```

Comment:
do we need to do this object persistent or something similar to avoid having issues with the garbage collector?